proto/desmos/profiles/v1beta1/models_app_links.proto

```diff
@@ -0,0 +1,136 @@
+syntax = "proto3";
+package desmos.profiles.v1beta1;
+
+option go_package = "github.com/desmos-labs/desmos/x/profiles/types";
+
+import "gogoproto/gogo.proto";
+import "google/protobuf/timestamp.proto";
+
+// ApplicationLink contains the data of a link to a centralized application
+message ApplicationLink {
+  option (gogoproto.equal) = true;
+
+  // Data contains the details of this specific link
+  LinkData data = 1
+      [ (gogoproto.nullable) = false, (gogoproto.moretags) = "yaml:\"data\"" ];
+  // Data used to verify the connection
+  VerificationData verification = 2 [
+    (gogoproto.nullable) = false,
+    (gogoproto.moretags) = "yaml:\"verification\""
+  ];
+  // State of the connection
+  ApplicationLinkState state = 3
+      [ (gogoproto.nullable) = false, (gogoproto.moretags) = "yaml:\"state\"" ];
+  // OracleRequest represents the request that has been made to the oracle
+  OracleRequest oracle_request = 4 [
+    (gogoproto.nullable) = false,
+    (gogoproto.moretags) = "yaml:\"oracle_request\""
+  ];
+  // Data coming from the result of the verification.
+  // Only available when the state is STATE_SUCCESS
+  ApplicationLinkResult result = 5 [ (gogoproto.moretags) = "yaml:\"result\"" ];
+  // CreationTime represents the time in which the link was created
+  google.protobuf.Timestamp creation_time = 6 [
+    (gogoproto.stdtime) = true,
+    (gogoproto.nullable) = false,
+    (gogoproto.moretags) = "yaml:\"creation_time\""
+  ];
+}
+
+// LinkData contains the data associated to a specific user of a generic
+// centralized application
+message LinkData {
```

Comment:
@dadamu @RiccardoM I would avoid naming something with the suffix `Data` and I think also the prefix `App` here is redundant.
I think `Link` is enough since it lives inside ApplicationLink and that's explicit enough to understand the meaning of it IMO. We should always avoid noise words like `Info` and `Data` where possible because they don't add any useful information at all.
(This is not something I've thought out of nowhere 🤣 I've read it in the clean code book from Robert C. Martins)

Reply:
You're right, I've change this to make it more explicit